Formula/gn.rb

```diff
@@ -0,0 +1,20 @@
+class Gn < Formula
+  desc "Generate Ninja - Chromium's build system"
+  homepage "https://gn.googlesource.com/gn/"
+  url "https://gn.googlesource.com/gn.git",
+    :revision => "f5fc06000379fe2c11fc130431e826e860ea6aae"
+  version "1547"
+  sha256 "8094c4ac075f918fdb28af10e16c9d4e41481702bd14a2c06cffd50fe03de9ba"
+  depends_on "ninja" => :build
+
+  def install
+    system "python", "build/gen.py"
+    system "ninja", "-C", "out/", "gn"
+    mkdir bin
+    cp "out/gn", bin/"gn"
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/gn --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
@SMillerDev PTAL, I have added a more robust test